File: 백준/Silver/11723. 집합/집합.py
import sys

M = int(input())
S = set()

for _ in range(M):
    op = sys.stdin.readline().rstrip().split()
    if len(op) == 1:
        if op[0] == "all":
            S = set([i for i in range(1, 21)])
        else:
            S = set()
    else:
        func, num = op[0], int(op[1])

        if func == "add":
            S.add(num)
        elif func == "remove":
            S.discard(num)
        elif func == "check":
            print(1 if num in S else 0)
        elif func == "toggle":
            if num in S:
                S.discard(num)
            else:
                S.add(num)

"""
26
add 1
add 2
check 1
check 2
check 3
remove 2
check 1
check 2
toggle 3
check 1
check 2
check 3
check 4
all
check 10
check 20
toggle 10
remove 20
check 10
check 20
empty
check 1
toggle 1
check 1
toggle 1
check 1

"""

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {

    static int stoi(String s) {
        return Integer.parseInt(s);
    }

    static int M;
    static Set<Integer> set = new HashSet<>();
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        M = stoi(st.nextToken());

        while (M-- > 0) {
            String[] arr = br.readLine().split(" ");

            String op = arr[0];
            int x;

            switch (op) {
                case "all":
                    set.clear();
                    for (int i = 1; i <= 20; i++) {
                        set.add(i);
                    }
                    break;

                case "empty":
                    set.clear();
                    break;

                case "add":
                    x = stoi(arr[1]);
                    set.add(x);
                    break;

                case "remove":
                    x = stoi(arr[1]);
                    set.remove(x);
                    break;

                case "toggle":
                    x = stoi(arr[1]);
                    if (set.contains(x)) {
                        set.remove(x);
                    } else {
                        set.add(x);
                    }
                    break;

                case "check":
                    x = stoi(arr[1]);
                    sb.append(set.contains(x) ? 1 : 0).append("\n");
            }
        }

        System.out.println(sb);
    }
}
